File: zad02/Projekt02/Projekt02/Program/Program/Wektor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Program
{
    class Wektor<M>:DzialaniaMatematyczne
    {
        public M[] wektor; //Deklaracja wektora okreslonego wczesniej typu

        public int wymiar;           //wymiar wektora
        static int max = 65535;     // 2^16 - 1
        static int min = -65536;    // -(2^16)
        static int dzielnik = 65536; // 2^16

        //Konstruktor
        public Wektor(int wymiar)
        {
            this.wymiar = wymiar;
            wektor = new M[wymiar];

        }


        //Funkcja losujaca macierz okreslonej wielkosci i zwracajaca ja
        //Wielkosc podajemy w konstruktorze 
        public void Losuj()
        {
          
            Random pseudolosowa = new Random();

            if (typeof(M) == typeof(float))//Generowanie liczb dla typu float
            {
                for (int i = 0; i < wymiar; i++)
                {
                    wektor[i] = (M)Convert.ChangeType(((float)pseudolosowa.Next(min, max) / (float)dzielnik), typeof(M));
                }
            }
            else
                 if (typeof(M) == typeof(double))//Generowanie liczb dla typu double
            {

                for (int i = 0; i < wymiar; i++)
                {
                    wektor[i] = (M)Convert.ChangeType(((double)pseudolosowa.Next(min, max) / (double)dzielnik), typeof(M));
                }
            }
 
        }

        public M[] GetWektor()
        {
            return wektor;
        }


        public void SkopiujDo(Wektor<M> w)
        {
       
                for (int i = 0; i < wymiar; i++)
                {
                    w.wektor[i] = wektor[i];
                }

        }


        public void SkopiujISkonwersujDo(Wektor<double> w)
        {
           
                for (int i = 0; i < wymiar; i++)
                {
                    w.wektor[i] = (double)Convert.ChangeType(wektor[i], typeof(double));
                }
            

        }
   


        public double WyliczanieDlugosciWektora()
        {
           
            M m = (M)Convert.ChangeType(0, typeof(M));
            for (int i = 0; i < wymiar; i++)
            {
                m = Add(m, Multiply(wektor[i], wektor[i]));
            }
            double wynik = (double)Convert.ChangeType(m, typeof(double));
            wynik = Math.Sqrt(wynik);
            return wynik;
        }


        public void WyswietlWektor()
        {

                for (int i = 0; i < wymiar; i++)
                {
                    Console.Write(wektor[i] + " ");
                }
                Console.WriteLine();

        }
    }
}

File: zad04/src/Wektor.java
import java.io.FileNotFoundException;
import java.io.PrintWriter;

public class Wektor {

    public double[] wektor;
    public int rozmiar;
    public Wektor(int rozmiar)
    {
        this.rozmiar = rozmiar;
        wektor = new double[rozmiar];
        WyzerujWektor();
        wektor[rozmiar - 1] = -1;
    }



    public Wektor(double [] wektor)
    {
        this.rozmiar = wektor.length;
        this.wektor = wektor;

    }


    public void WyzerujWektor()
    {
        for (int i = 0; i < rozmiar; i++)
            wektor[i] = 0;
    }

    public  void WyswietlWektor()
    {
        for(int i=0;i<rozmiar;i++)
        {
            System.out.println(wektor[i]);
        }

    }

    public void SkopiujDo(Wektor w)
    {
        for (int i = 0; i < rozmiar; i++)
            w.wektor[i] = (double)this.wektor[i];
    }

    public double ObliczBladZ(Wektor w)
    {
        double wynik;
        Wektor pom = new Wektor(rozmiar);
        pom.WyzerujWektor();

        for(int i=0;i<rozmiar;i++)
        {
            pom.wektor[i] = wektor[i] - w.wektor[i];
            //  Console.WriteLine(pom.wektor[i] + "   " + wektor[i] + "   " + w.wektor[i]);
            if (pom.wektor[i] < 0)
                pom.wektor[i] *= (double)-1;
        }

        wynik = pom.wektor[0];
        for(int i=1;i<rozmiar;i++)
        {
            if (pom.wektor[i] > wynik)
                wynik = pom.wektor[i];

        }


        return wynik;
    }

    public double WyliczanieDlugosciWektora(Wektor w2)
    {

        double m = 0;
        double m2 = 0;
        for (int i = 0; i < rozmiar; i++)
        {
            m += wektor[i]* wektor[i];
            m2 += w2.wektor[i] * w2.wektor[i];
        }
        double wynik;
        double wynik2;
        wynik = Math.sqrt(m);
        wynik2 = Math.sqrt(m2);

        if (wynik - wynik2 > 0)
            return wynik - wynik2;
        else
            return (-1) * (wynik - wynik2);
    }

    public void ZapiszDoPliku(String nazwa) throws FileNotFoundException {
        PrintWriter zapis = new PrintWriter(nazwa);
        String text="";

        for (int i=0;i<rozmiar;i++)
        {
            text = String.valueOf(wektor[i]) + "\n";

            zapis.print(text);
        }

        zapis.print("\r\n");
        zapis.print("\r\n");
        zapis.close();
    }
}
